src/app/core/services/label.service.ts

```diff
@@ -0,0 +1,136 @@
+import { Injectable } from '@angular/core';
+import {GithubService} from './github.service';
+import {map} from 'rxjs/operators';
+import {Label} from '../models/label.model';
+import { Observable } from 'rxjs';
+import {SEVERITY_ORDER} from '../../core/models/issue.model';
+import { User } from '../models/user.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LabelService {
+  private severityLabels: Label[];
+  private typeLabels: Label[];
+  private responseLabels: Label[];
+  private duplicateLabelMap: Map<string, boolean>;
+
+
+  constructor(private githubService: GithubService) {
+    this.severityLabels = new Array();
+    this.typeLabels = new Array();
+    this.responseLabels = new Array();
+    this.duplicateLabelMap = new Map();
+  }
+
+  /**
+   * Calls the Github service api to get all labels from the repository and
+   * store it in a list of arrays in this label service
+   */
+  getAllLabels(userResponse: User): Observable<User> {
+      return this.githubService.fetchAllLabels().pipe(
+        map((response) => {
+          this.populateLabelLists(response);
+          return userResponse;
+        })
+      );
+  }
+
+  /**
+   * Get all the labels of a certain type (e.g severity)
+   * @param attributeName: the type of the label
+   * @return an array of label of that type
+   */
+  getLabelList(attributeName: string): Label[] {
+    switch (attributeName) {
+      case 'severity':
+        return this.severityLabels;
+      case 'type':
+        return this.typeLabels;
+      case 'responseTag':
+        return this.responseLabels;
+    }
+  }
+
+  /**
+   * Store the json data from Github api into the list of arrays in this service
+   * @param labels: the json data of the label
+   */
+  private populateLabelLists(labels: Array<{}>): void {
+
+    for (const label of labels) {
+      // Get the name and color of each label and store them into the service's array list
+      const labelName = String(label['name']).split('.');
+      const labelType = labelName[0];
+      const labelValue = labelName[1];
+      const labelColor = String(label['color']);
+
+      if (!this.duplicateLabelMap.has(labelValue)) {
+        this.duplicateLabelMap.set(labelValue, true);
+
+        switch (labelType) {
+          case 'severity':
+            this.severityLabels.push({labelValue: labelValue, labelColor: labelColor});
+            break;
+          case 'type':
+            this.typeLabels.push({labelValue: labelValue, labelColor: labelColor});
+            break;
+          case 'response':
+            this.responseLabels.push({labelValue: labelValue, labelColor: labelColor});
+            break;
+        }
+      }
```

Comment:
convert this to a guard condition to reduce nesting?
`if (!this.duplicateLabelMap.has(labelValue)) {`
->
```
if (this.duplicateLabelMap.has(labelValue)) {
    continue;
}
...
```